File: 조민준/2기/1주차/푸드파이트대회.java
class 푸드파이트대회 {
    /*
     * 데칼코마니식의 출력 결과를 내면 되는 문제로
     * left, right 로 나눠서 반복문을 통해 결과를 출력했다.
     */
	public String solution(int[] food) {
      
        String answer = "";
        String left = "";	// 정방향 변수
        String right = "";	// 역방향 변수
        
        // 반복문1 - 정방향
        for(int i = 1; i < food.length; i++) {
            for(int j = 0; j < food[i] / 2; j++) {
                left += i;
            }
        }
        
        // 반복문2 - 역방향
        for(int i = food.length -1; i > 0; i--) {
        	for(int j = 0; j < food[i] /2; j++) {
        		right += i;
        	}
        }
        // 출력값 = 정방향 + '0' + 역방향
        answer = left + '0' + right;
        return answer;
    }
}

File: 이준혁/2기/1주차/푸드파이트대회.js
function solution(food) {
    // 푸드 파이트 대회는 1:1
    // 한 선수는 왼쪽부터 오른쪽으로
    // 다른 선수는 오른쪽에서 왼쪽으로 먹음
    // 칼로리가 낮은 음식을 먼저 먹을 수 있게 배치
    // 물 (0)을 먼저 먹으면 승리
    let result = ''; // 빈 문자열 생성

    for (let i = 1; i < food.length; i++) { // food[0]의 경우 물이기에 제외하고, 나머지를 반복하는데
        if (food[i] % 2 === 0) { // 만약 2로 나눠떨어진다면 
            for (let j = 0; j < food[i] / 2; j++) { // 절반이기에 모두 사용하므로 절반만큼을 반복하여
                result += i; // 결과값에 해당 인덱스를 추가해준다
            }
        } else { // 2로 떨어지지 않는다면 (홀수)
            for (let j = 1; j < food[i] / 2; j++) { // 반복을 한 번 덜하여
                result += i; // 결과값에 해당 인덱스를 추가해준다.
            }
        } 
    }

    let tmp = result.split(''); // split을 이용하여 배열단위로 저장한 값을 tmp에 넣어주고
    tmp = tmp.reverse(); // reverse로 반대로 뒤집어준다.
    tmp = tmp.join(''); // 그 값을 join을 이용하면 합칠 수 있다.

    result += 0; // 가운데에 물의 값인 0을 넣어주고
    result += tmp; // tmp을 붙여주면 정답

    return result;
}
